_data/data.js

```diff
@@ -7,6 +7,19 @@ const Parser = require('rss-parser');
 const Axios = require('axios');
 const Sharp = require('sharp');
 
+const uriJoin = (...paths) => {
+  if (!Array.isArray(paths))
+    return new TypeError('TypeError: uriJoin must have at least one param');
+  const resolvedUrl = [].reduce.call(paths, (prev, curr) => {
+    return new URL(curr, new URL(prev + '/', 'resolve://'));
```

Comment:
We need to give some special treatment to that extra `'/'` you append to `prev`. I understand why you did that -- when `prev` doesn't end with a `/` the last segment of `prev` will be replaced by `curr`. That was actually the intended behavior of `url.resolve`:

```
url.resolve('/one/two/three', 'four');         // '/one/two/four'
```

We can see that `three` is replaced by `four` instead of becoming `three/four`. We don't want exactly the same behavior. However, we need to deal with this case:

```
uriJoin('/one/', 'two');         // '/one//two'
```

We should use use `prev.endsWith('/')` to determine whether we want to append another `/`.

```suggestion
    return new URL(curr, new URL(prev.endsWith('/') ? prev : prev + '/', 'resolve://'));
```